Sources/LanguageServerProtocol/SupportTypes/CompletionItem.swift

```diff
@@ -11,7 +11,23 @@
 //===----------------------------------------------------------------------===//
 
 /// A single completion result.
-public struct CompletionItem: Codable, Hashable {
+public struct CompletionItem: TextDocumentRequest, RequestType, ResponseType, Codable, Hashable {
+  public typealias Response = CompletionItem
+    
+  public var textDocument: TextDocumentIdentifier {
+    if 
+      let data = self.data,
+      case let .dictionary(dict) = data,
+      case let .string(textDocURI) = dict["textDocURI"]
+    {
+      return TextDocumentIdentifier(DocumentURI(string: textDocURI))
+    } else {
+      fatalError("Missing textDocURI in CompletionItem.data: \(data ?? "(CompletionItem.data is nil)")")
```

Comment:
I don’t think we should make any assumptions about the requests sent to us by the client and, in particular, not crash if it is malformed. Could we return a failure value (e.g.`nil`) here instead of `fatalError`ing?